share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
I was just looking at Cygwin headers related to this. I can kind of see which macros guard them, but it is totally not clear that the same macros are not used at some other key locations since they do not seem to be pthreads-specific.
Here is another proposition - Monty, who most likely will not like this idea :), can try to rewrite our model so that nothing gets stored in C structures. Basically, all state would be tracked using custom Boogie maps. I think that should be doable, and I also think that it should be portable since we would not have to redefine C types - I think. There are two drawbacks that I can think of: rewriting everything is probably a good amount of work and creating all these maps within Boogie might potentially prevent some opportunities for scalarization since DSA would not kick in.
Other than that. I am out of ideas. Thoughts?